apps/cookbook/src/app/page/side-nav/side-nav.component.ts

```diff
@@ -1,20 +1,160 @@
+  path: string;
+  name: string;
+  active: boolean;
+  hidden: boolean;
+}
 
 @Component({
   selector: 'cookbook-side-nav',
   templateUrl: './side-nav.component.html',
   styleUrls: ['./side-nav.component.scss'],
 })
 export class SideNavComponent implements OnInit {
+  private allShowcaseRoutes: ISideNavLink[];
+  filteredShowcaseRoutes: ISideNavLink[];
+  filter: string = '';
+
   @Output() menuToggle = new EventEmitter<boolean>();
   @Input() isMenuOpen = false;
 
-  constructor() {}
+  constructor(private router: Router) {}
+
+  navigationLinks = navigationItems;
+
+  ngOnInit() {
+    this.mapRoutes();
+    this.filteredShowcaseRoutes = this.allShowcaseRoutes;
+    this.router.events.subscribe((val) => {
+      if (val instanceof NavigationEnd) {
+        if (val.url.indexOf('showcase') === -1) {
+          // reset component links
+          this.filterComponents();
+        }
+      }
+    });
+  }
+
+  private mapRoutes() {
+    const routesWithPath = routes[0].children.filter((r) => r.path);
+    routesWithPath.sort((a, b) => {
+      return a.path < b.path ? -1 : a.path > b.path ? 1 : 0;
+    });
+    this.allShowcaseRoutes = routesWithPath.map((route) => {
+      return {
+        path: `showcase/${route.path}`,
+        name: this.convertKebabToTitleCase(route.path),
+        active: this.router.url.indexOf(route.path) > -1,
+        hidden: false,
+      };
+    });
+  }
+
+  @ViewChildren('componentLink') componentLinks: QueryList<any>;
+
+  onFilterChange(event) {
+    this.filter = event;
+    this.filterComponents();
+  }
+
+  onFilterKeyDown(event: KeyboardEvent) {
+    if (event.key === KEY_DOWN) {
+      event.preventDefault();
+      const links = this.componentLinks.toArray();
+      for (let i = 0; i < links.length; i++) {
+        const element = links[i].nativeElement;
+        if (!element.classList.contains(HIDDEN_CLASS)) {
+          element.focus();
+          break;
+        }
+      }
+    }
+  }
+
+  onLinksKeyDown(event: KeyboardEvent) {
+    if (event.key === KEY_DOWN || event.key === KEY_UP) {
+      event.preventDefault();
+    } else {
+      return;
+    }
 
-  ngOnInit() {}
+    const visibleLinks = this.componentLinks.filter((link) => {
+      return !link.nativeElement.classList.contains(HIDDEN_CLASS);
+    });
+    const currentlyFocused = visibleLinks.findIndex((link) => {
+      return link.nativeElement === document.activeElement;
+    });
+    if (currentlyFocused === -1) {
+      return;
+    }
+
+    if (event.key === KEY_DOWN) {
+      visibleLinks[Math.min(currentlyFocused + 1, visibleLinks.length - 1)].nativeElement.focus();
+    } else {
+      visibleLinks[Math.max(currentlyFocused - 1, 0)].nativeElement.focus();
+    }
+  }
+
+  onComponentClick(event) {
+    this.setRouteActive(event.path);
+    this.onToggleMenu();
+  }
 
   onToggleMenu() {
     this.isMenuOpen = !this.isMenuOpen;
     this.menuToggle.emit(this.isMenuOpen);
   }
+
+  private convertKebabToTitleCase(kebab) {
```

Comment:
Consider extracting to/aligning with https://github.com/kirbydesign/designsystem/blob/master/libs/designsystem/src/lib/helpers/color-helper.ts#L133
(Maybe move somewhere else, i.e. a `LanguageHelper` or `LanguageUtil` or similar...)

Reply:
Decided on a string-helper.ts